### PR TITLE
Endpoints for searching resources

### PR DESCRIPTION
**Pull request by krisx303**

### CODE REVIEW

File: src/main/java/com/brightpath/learnify/controller/QuizController.java [hunk 1]
```diff
@@ -151,9 +154,11 @@ public ResponseEntity<QuizResultUpdateDto> updateResultsByQuizId(UUID quizId, Qu
     }
 
     @Override
-    public ResponseEntity<List<QuizSummaryDto>> listQuizzes(UUID workspaceId) {
+    public ResponseEntity<List<QuizSummaryDto>> listQuizzes(@Nullable String name, @Nullable String ownerId, @Nullable ResourceAccessTypeDto accessType, @Nullable UUID workspaceId) {
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/controller/QuizController.java:159]: thanks for adding workspaceId also

File: src/main/java/com/brightpath/learnify/controller/mapper/DtoMapper.java [hunk 2]
```diff
@@ -58,6 +59,7 @@ public NoteSummaryDto asNoteSummaryDto(Note note) {
                 .createdAt(note.createdAt())
                 .type(asNoteTypeDto(note.type()))
                 .pagesCount(note.pagesCount())
+                .accessType(toResourceAccessTypeDto(note.permissionLevel()))
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/controller/mapper/DtoMapper.java:66]: we do not want optionals?

File: src/main/java/com/brightpath/learnify/persistance/note/NoteRepository.java [hunk 3]
```diff
@@ -14,7 +15,18 @@ public interface NoteRepository extends JpaRepository<NoteEntity, UUID> {
     @Query("""
             SELECT u
             FROM NoteEntity u
-            WHERE (:workspaceId IS NULL OR u.workspace.id = :workspaceId)
+            JOIN PermissionsAccessEntity access ON u.id = access.resourceId
```
Comment (Adam3004) [inline src/main/java/com/brightpath/learnify/persistance/note/NoteRepository.java:18]: nice

File: src/main/resources/api.yaml [hunk 4]
```diff
@@ -301,14 +318,31 @@ paths:
       tags:
         - quizzes
       parameters:
+        - name: name
```
Comment (Adam3004) [inline src/main/resources/api.yaml]: if we use it twice maybe define it below as other params?

**Review by Adam3004 — APPROVED** [review]

File: src/main/java/com/brightpath/learnify/controller/mapper/DtoMapper.java (hunk 2 above)
Comment (krisx303) [reply src/main/java/com/brightpath/learnify/controller/mapper/DtoMapper.java:66]: I don't think Optional would be useful here. We expect nullable value in controller and JPA query

**Pull request merged**